src/main/java/com/snowflake/kafka/connector/internal/SnowflakeSinkServiceV1.java

```diff
@@ -1085,22 +1086,27 @@ private void filterResultFromSnowpipeScan(
 
     private void purge(List<String> files) {
       if (!files.isEmpty()) {
-        LOGGER.debug(
-            "Purging loaded files for pipe:{}, loadedFileCount:{}, loadedFiles:{}",
-            pipeName,
-            files.size(),
-            Arrays.toString(files.toArray()));
+        String logString = String.format(
+            "Purging loaded files for pipe: %s, loadedFileCount: %d", pipeName, files.size()
+        );
+        LOGGER.info(logString + prepareFilesOffsetsLogString(
+                files, "loadedFiles", LOGGER.isDebugOrTraceEnabled()
+            )
+        );
         conn.purgeStage(stageName, files);
       }
     }
 
     private void moveToTableStage(List<String> failedFiles) {
       if (!failedFiles.isEmpty()) {
-        LOGGER.debug(
-            "Moving failed files for pipe:{} to tableStage failedFileCount:{}, failedFiles:{}",
-            pipeName,
-            failedFiles.size(),
-            Arrays.toString(failedFiles.toArray()));
+        String logString = String.format(
+            "Moving failed files for pipe: %s to tableStage failedFileCount: %d",
+            pipeName, failedFiles.size()
+        );
+        LOGGER.info(logString + prepareFilesOffsetsLogString(
+                failedFiles, "failedFiles", LOGGER.isDebugOrTraceEnabled()
```

Comment:
it should be logger responsibility to decide when to display certain logs. Please do not build strings based on log level. Let's split them into two separate logs, one will be passed to LOGGER.info() and second one, more detailed to LOGGER.debug()